### PR TITLE
Refactor PD member count in failover

### PR DESCRIPTION
**Pull request by handlerww**
### What problem does this PR solve?
Fix member count according to comments in pkg/manager/member/pd_failover.go in #3550 

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/manager/member/pd_failover.go [hunk 1]
```diff
@@ -63,20 +63,15 @@ func (f *pdFailover) Failover(tc *v1alpha1.TidbCluster) error {
 			ns, tcName, healthCount, tc.PDStsDesiredReplicas(), tc.Spec.PD.Replicas, len(tc.Status.PD.FailureMembers))
 	}
 
-	failureReplicas := tc.GetPDFailureReplicas()
-	if failureReplicas >= *tc.Spec.PD.MaxFailoverCount {
-		klog.Errorf("PD failover replicas (%d) reaches the limit (%d), skip failover", failureReplicas, *tc.Spec.PD.MaxFailoverCount)
+	deletedFailureReplicas := tc.GetPDFailureReplicas()
```
Comment (csuzhangxc) [inline pkg/manager/member/pd_failover.go:66]: How about giving `GetPDFailureReplicas` a better name?

Comment (handlerww) [reply pkg/manager/member/pd_failover.go:66]: We can use CountPDFailureReplicasTobeDeleted. I will update the changes.

Comment (handlerww) [reply pkg/manager/member/pd_failover.go]: fixed

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Comment by DanielZhangQD** [comment]
@handlerww Please resolve the conflict.

File: pkg/apis/pingcap/v1alpha1/tidbcluster.go [hunk 2]
```diff
@@ -324,7 +324,7 @@ func (tc *TidbCluster) PDAutoFailovering() bool {
 	return false
 }
 
-func (tc *TidbCluster) GetPDFailureReplicas() int32 {
+func (tc *TidbCluster) CountPDFailureReplicasTobeDeleted() int32 {
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/tidbcluster.go:327]:
```suggestion
func (tc *TidbCluster) GetPDDeletedFailureReplicas int32 {
```

File: pkg/manager/member/pd_failover.go [hunk 3]
```diff
@@ -63,21 +63,16 @@ func (f *pdFailover) Failover(tc *v1alpha1.TidbCluster) error {
 			ns, tcName, healthCount, tc.PDStsDesiredReplicas(), tc.Spec.PD.Replicas, len(tc.Status.PD.FailureMembers))
 	}
 
-	failureReplicas := tc.GetPDFailureReplicas()
-	if failureReplicas >= *tc.Spec.PD.MaxFailoverCount {
-		klog.Errorf("PD failover replicas (%d) reaches the limit (%d), skip failover", failureReplicas, *tc.Spec.PD.MaxFailoverCount)
+	pdFailureReplicasTobeDeleted := tc.CountPDFailureReplicasTobeDeleted()
```
Comment (DanielZhangQD) [inline pkg/manager/member/pd_failover.go:66]: `pdDeletedFailureReplicas`

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**